core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -947,7 +945,9 @@ private[spark] object Utils extends Logging {
    */
   def isSymlink(file: File): Boolean = {
     if (file == null) throw new NullPointerException("File must not be null")
-    if (isWindows) return false
+    if (isWindows) {
+      return Files.isSymbolicLink(Paths.get(file.toURI))
```

Comment:
Let's just invoke this method as the body of this method, now that we can use Java 7, for Windows and all platforms.